src/main/java/com/iprd/fhir/utils/FhirUtils.java

```diff
@@ -53,6 +52,36 @@ public static Triple<String, String, String> getOclIdFromIdentifier(List<Identif
 		return oclId;
 	}
 
+	public static Pair<String, String> getEncounterIdAndOrganizationIdForAppointment(String appointmentId, IGenericClient fhirClient){
+		Bundle appointmentBundle = fhirClient.search().forResource(Appointment.class).where(Appointment.RES_ID.exactly().identifier(appointmentId)).returnBundle(Bundle.class).execute();
+		if (!appointmentBundle.hasEntry()){
+			return null;
+		}
+		Appointment appointment = (Appointment) appointmentBundle.getEntry().get(0).getResource();
+		String questionnaireResponseId = ((Reference) appointment.getExtensionByUrl("http://iprdgroup.com/fhir/StructureDefinition/resource-questionnaireResponse").getValue()).getReferenceElement().getIdPart();
```

Comment:
Please use the Provenance to get the questionnaire response id. This extension is no longer used.